Services/UI/OJS.Services.Ui.Data/Implementations/ParticipantsDataService.cs

```diff
@@ -25,17 +25,35 @@ public ParticipantsDataService(OjsDbContext db)
             => this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
                 .FirstOrDefaultAsync();
 
-        public Task<Participant?> GetWithContestByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial)
+        public Task<Participant?> GetWithContestAndSubmissionDetailsByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial)
             => this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
                 .Include(p => p.Contest)
-                .ThenInclude(c => c.ProblemGroups)
-                .ThenInclude(pg => pg.Problems)
+                    .ThenInclude(c => c.ProblemGroups)
+                        .ThenInclude(pg => pg.Problems)
+                            .ThenInclude(p => p.SubmissionTypesInProblems)
+                                .ThenInclude(sp => sp.SubmissionType)
+                .Include(p => p.Scores)
+                .Include(p => p.ProblemsForParticipants)
+                .FirstOrDefaultAsync();
+
+        public Task<Participant?> GetWithProblemsForParticipantsByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial)
+            => this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
+                .Include(p => p.ProblemsForParticipants)
+                    .ThenInclude(pfp => pfp.Problem)
                 .FirstOrDefaultAsync();
 
         public IQueryable<Participant> GetAllByUser(string? userId)
             => this.DbSet
                 .Where(p => p.UserId == userId);
 
+        public IQueryable<Participant> GetAllWithContestSubmissionsAndProblemsByUser(string? userId)
+            => this.DbSet
+                .Where(p => p.UserId == userId)
+                .Include(p => p.Submissions)
```

Comment:
Do you need all the `.Include`s, I see that you use MapCollection<> on this `IQueryable<Participant>`

Reply:
Method has been double testest and the .Include for Submissions has been removed. The contest includes are need for the further mapping